### DISCUSSION TITLE
Configurable config timeout

### DISCUSSION THREAD

**Discussion by atunguyd** [universam1/iSpindel#587] [Ideas]
Hi Can I suggest  a feature for the code?

Can we not configure how long the timeout for config mode is? I have found that when calibrating my iSpindle it is easier to log into the config page and look at the detected angle there while I am changing the density of the sugar water solution. However the config page often times out and I need to open the iSpindel and hit reset three times.

I have resolved this with a local build where I simple editted globals.h and changed #define PORTALTIMEOUT 300 to 3600
It would be better if it were possible to have an override in the config pages of the user interface where someone can select this timeout and it is preserved through reboots.

I am happy to make this change if someone could point to me where the UI is managed. My skills are all embedded low level drivers and the like, not UI and http interfaces.

**Comment by universam1** [comment]
The config mode is a power drain. In order to prevent the battery to be drained by accident this timeout is in place and necessary.

**Reply by atunguyd** [reply]
I understand that - am I correct in saying that the only way to accidently go into config mode is to press reset three times? Looking at the code it looks like that is the case to me.  So the chances of going into config mode while fermenting is very small is it not?
For me I see a  big benefit in using the config pages for the calibration procedure.  This is much faster than waiting for the iSpindel to sync to ubidots for just one measurement.
 Having the timeout user adjusted would not cause battery drain in normal fermenting mode, unless you see a situation where, during a brew, the iSpindel would go into configuration mode? 

Already for the calibration procedure we adjust the ubidots reporting time, so for calibration we need to go into config and change things anyway - it would be easier to rather change the config timeout and then use the config page for calibration which is a much faster process than using ubidots. 

I hope what I am saying makes sense? I am referring to being able to use this page at http://192.168.4.1/iSpindel?  for calibration:
![image](https://user-images.githubusercontent.com/12063824/181173667-32056527-6669-4edb-aafe-598fef3d03e4.png)
